### PR TITLE
Fix backward Relay pagination

### PR DESCRIPTION
**Pull request by tcleonard**
New version of the pull request #1010 (to solve issue #1009) which adds unit tests.

### CODE REVIEW

**Comment by tcleonard** [comment]
Not sure what is causing those test failures... it fails in an import in django rest framework (and the import is totally valid).
Tests are passing locally.

**Comment by zbyte64** [comment]
This is the relevant part I see from the test failure:

```
 .tox/py37-django111/lib/python3.7/site-packages/rest_framework/fields.py:15: in <module>
      from django.core.validators import (
  E   ImportError: cannot import name 'ProhibitNullCharactersValidator' from 'django.core.validators' (/home/runner/work/graphene-django/graphene-django/.tox/py37-django111/lib/python3.7/site-packages/django/core/validators.py)
```

DRF wants to import a validator from Django that doesn't exist in Django 1.11

**Comment by tcleonard** [comment]
Indeed, which is not related to anything I have changed... Maybe DRF simply decided to drop support for older versions of Django and we might need to do the same here.

**Comment by jkimbo** [comment]
@tcleonard yep DRF dropped support for Django 1.11. Fixed the tests here: https://github.com/graphql-python/graphene-django/pull/1047

**Comment by tcleonard** [comment]
thanks @jkimbo, merged your changes and the CI now passes

**Review by zbyte64 — APPROVED** [review]
Thank you for this

**Comment by vecchp** [comment]
@zbyte64 is there anything that needs to be done to get this in?

**Comment by tcleonard** [comment]
I don't think so... for me it's ready to merge.. 🤷

File: graphene_django/fields.py [hunk 1]
```diff
@@ -135,22 +135,23 @@ def resolve_connection(cls, connection, args, iterable, max_limit=None):
...
 
         # If after is higher than list_length, connection_from_list_slice
         # would try to do a negative slicing which makes django throw an
         # AssertionError
         after = min(get_offset_with_default(args.get("after"), -1) + 1, list_length)
 
         if max_limit is not None and "first" not in args:
-            args["first"] = max_limit
+            if "last" in args:
+                args["first"] = list_length
```
Comment (pcraciunoiu) [inline graphene_django/fields.py:151]: Wondering why is this needed? In #1074 I didn't have to include it, seems like removing it wouldn't change anything.

File: graphene_django/tests/test_query.py [hunk 2]
```diff
@@ -1213,6 +1213,103 @@ class Query(graphene.ObjectType):
...
+    def test_should_query(self, graphene_settings):
+        """
+        Backward pagination should work as expected
+        """
+        schema = self.setup_schema(graphene_settings, max_limit=100)
+        self.do_queries(schema)
+
+    def test_should_query_with_low_max_limit(self, graphene_settings):
+        """
+        When doing backward pagination (using last) in combination with a max limit higher than the number of objects
```
Comment (pcraciunoiu) [inline graphene_django/tests/test_query.py:1306]: In this case, max limit is lower than the number of objects, right?

It looks like this test covers the same case as [this one](https://github.com/graphql-python/graphene-django/pull/1074/files#diff-cb65a9e9b2bd8192f5623c19ed4124223b411da82d4daf9770910074673adbb4R730), which is great!

**Review by pcraciunoiu — APPROVED** [review]
Makes sense to me. Just one change I'd suggest.

**Comment by pcraciunoiu** [comment]
@jkimbo any chance we could get this merged?

Because graphene-django doesn't support true cursor pagination, the most reliable way to get recent items first is to go backwards, and this bug gets in the way of that.

**Comment by jkimbo** [comment]
Thanks @zbyte64

File: graphene_django/fields.py (hunk 1 above)
Comment (tcleonard) [reply graphene_django/fields.py:151]: I would have to check if it is actually used by `connection_from_list_slice` but I set it at least for consistency with the other case where it is set to `max_limit`

**Comment by tcleonard** [comment]
Thanks @zbyte64
